build-system/tasks/sweep-experiments/index.js

```diff
@@ -0,0 +1,445 @@
+  files.map((path) => `- \`${path}\``).join('\n');
+
+/**
+ * @return {string}
+ */
+const readmeMdGithubLink = () =>
+  `https://github.com/ampproject/amphtml/${path.relative(
+    process.cwd(),
+    __dirname
+  )}`;
+
+/**
+ * @param {{
+ *   removed: string,
+ *   cutoffDateFormatted: string,
+ *   modifiedSourceFiles: Array<string>,
+ *   htmlFilesWithReferences: Array<string>,
+ * }} vars
+ * @return {string}
+ */
+function summaryCommitMessage({
+  removed,
+  cutoffDateFormatted,
+  modifiedSourceFiles,
+  htmlFilesWithReferences,
+}) {
+  const paragraphs = [
+    `🚮 Sweep experiments older than ${cutoffDateFormatted}`,
+    `Sweep experiments last flipped globally up to ${cutoffDateFormatted}:`,
+    removed.join('\n'),
+  ];
+
+  if (modifiedSourceFiles.length > 0) {
+    paragraphs.push(
+      '---',
+      '### ⚠️ Javascript source files require intervention',
+      'The following may contain errors and/or require intervention to remove superfluous conditionals:',
+      fileListMarkdown(modifiedSourceFiles),
+      `Refer to the removal guide for [suggestions on handling these modified Javascript files.](${readmeMdGithubLink()}#followup)`
+    );
+  }
+
+  if (htmlFilesWithReferences.length > 0) {
+    paragraphs.push(
+      '---',
+      '### ⚠️ HTML files may still contain references',
+      'The following HTML files contain references to experiment names which may be stale and should be manually removed:',
+      fileListMarkdown(htmlFilesWithReferences),
+      `Refer to the removal guide for [suggestions on handling these HTML files.](${readmeMdGithubLink()}#followup:html)`
+    );
+  }
+
+  return paragraphs.join('\n\n');
+}
+
+/**
+ * @param {!Object<string, *>} prodConfig
+ * @param {!Object<string, *>} canaryConfig
+ * @param {string} cutoffDateFormatted
+ * @param {string=} removeExperiment
+ * @return {!Object<string, {percentage: number, previousHistory: Array}>}
+ */
+function collectWork(
+  prodConfig,
+  canaryConfig,
+  cutoffDateFormatted,
+  removeExperiment
+) {
+  if (removeExperiment) {
+    // 0 if not on prodConfig
+    const percentage = Number(prodConfig[removeExperiment]);
+    const previousHistory = findConfigBitCommits(
+      cutoffDateFormatted,
+      prodConfigPath,
+      removeExperiment,
+      percentage
+    );
+    return {[removeExperiment]: {percentage, previousHistory}};
+  }
+
+  const work = {};
+  for (const [experiment, percentage] of Object.entries(prodConfig)) {
+    if (
+      typeof percentage === 'number' &&
+      percentage === canaryConfig[experiment] &&
+      experiment !== 'canary' &&
+      (percentage >= 1 || percentage <= 0)
+    ) {
+      const previousHistory = findConfigBitCommits(
+        cutoffDateFormatted,
+        prodConfigPath,
+        experiment,
+        percentage
+      );
+      if (previousHistory.length > 0) {
+        work[experiment] = {percentage, previousHistory};
+      }
+    }
+  }
+  return work;
+}
+
+async function sweepExperiments() {
```

Comment:
Maintainability suggestion: add JSDoc above this function indicating that it's an entry point to a gulp task.